### ISSUE TITLE
COPYING license is inconsistent with 924 forks; please change to Attribution-NonCommercial-sharealike

### ISSUE THREAD

**Issue by simsong** [emeryberger/CSrankings#2338]
The file [COPYING](https://github.com/emeryberger/CSrankings/blob/gh-pages/COPYING) specifies the license **Attribution-NonCommercial-NoDerivatives 4.0 International** but the git repo has been forked 924 times (at the time of this writing). 

These would appear to be in conflict. If I fork the repo, can I make changes to the software, or is that prohibited under the license? If so, why does the repo allow forks? 

I request that the license be changed to Attribution-NonCommercial-sharealike.

**Comment by emeryberger** [comment]
Forks are for pull requests, and they are almost entirely for updates to the data. This is compatible with the license and I am not changing it.

**Comment by simsong** [comment]
Okay. Are you claiming copyright on the submitted data?

**Comment by emeryberger** [comment]
I am not a lawyer and do not know what the implications would be one way or the other.

**Comment by simsong** [comment]
Thank you for your willingness to engage with me on this topic. 

Your repo consists of both sofware and text. The use of the CC license for software is unusual *Attribution-NonCommercial-NoDerivatives 4.0 International* because you are literally telling students and other faculty members that they may download your code but that they may not make derrivative works and show them to others. So this prohibits people from producing and posting detailed visualizations using your data—for example, performing an analysis that crosses your data with those from the [CRA Taulbee Survey](https://cra.org/resources/taulbee-survey/).

The CC-NC-ND license does not permit people who download your code from fixing bugs or adding new features, bedcause both are considered derivatives. This is why it is not listed as an open source license in the [list of licenses maintained by the Open Source Initiative](https://opensource.org/licenses/alphabetical). 

US Copyright Law generally olds that it is not permissable to copyright a database, as these are seen as collections of facts that do not have creative input. That is, US law does not grant copyright under the "sweat of the brow" doctrine. [The Database and Collections of Information Misappropriation Act of 2003](https://en.wikipedia.org/wiki/Database_and_Collections_of_Information_Misappropriation_Act) was an attempt change the law, but the proposed act never became a law. If it had, it might have had a hard time in the courts, as Article I grants Congress the power to issue copyrights and patents to "authors and inventors," and it is unclear if assembling a database would be considered authoriship under the current body of case law. 

I filed this issue with you because I think that CC-NC-ND is inconsistent with both your stated goals of this project and with the way that you are implementing it. I think that the code should be covered under one of the open source licenses—your intent would probably be best served by the GNU Copyleft—and I think that the contributions of the 900+ contributors to your database would be more appropriately licensed under the CC-BY-SA or the [GNU Free Documentation License](https://en.wikipedia.org/wiki/GNU_Free_Documentation_License). 

For background, I recommend the excellent [FAQ on CC Licenses](https://creativecommons.org/faq/) that Creative Commons has produced. 


I do not think that there would be a legal issue in changing your license to a less restrictive and (in my opinion) a more academically appropriate license, because I do not think that the contributions of the 900+ academics meets the standard of a copyrightable works under US Law. And, as I noted in my original issue, *if they do meet the standard of copyrightable works, then they are derrivative works and they violate the copyright license that you have established on this git repo.*

This is certainly a complex area and I am not a lawyer, but I am familiar with these issues. If you wish, I can ask one of my colleagues who is an attorney to weigh in. 

I think that your project is important is clearly having an impact. However, I think that the licensing choice that you made has the impact of putting students, academics, and other researchers that want to build upon what you have done in legal jeopordy, which I do not think is your intent.

**Comment by emeryberger** [comment]
Thanks for the details. It's indeed a complex area that I am strongly disinclined to wade into further. I chose this license specifically because it prohibits mashups, which is something I explicitly do not want to allow (for a variety of reasons). I have granted permission upon request for a number of academic uses of CSrankings (e.g., data analyses). So far, this has not proven to be a problem, and as a practical person, I do not feel like addressing non-problems. If you have a specific request and not a blanket change based on a hypothetical, I'd be happy to discuss it over e-mail. Again, IANAL and I have zero interest in entering into a legal morass.
